### PR TITLE
Refactoring helper functions

### PR DESCRIPTION
**Pull request by shubhscoder**
In order to avoid using id/username in the endpoint some refactoring was needed in the helper functions
This PR does that refactoring

### CODE REVIEW

**Comment by shubhscoder** [comment]
@Radhikadua123 please review.

File: test/unit/utils_test.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 class UtilsTest < ActiveSupport::TestCase
   include Utils
-
+
```
Comment (Radhikadua123) [inline test/unit/utils_test.rb]: We really need to setup overcommit in our project. I have filed an issue for it. https://github.com/publiclab/plots2/issues/3967

But never mind, it's fine to commit it like this for now.

File: app/models/user.rb [hunk 2]
```diff
@@ -419,17 +419,20 @@ def customize_digest(type)
...
-  def validate_token(token)
-    decrypted_data = decrypt(token)
-    if id != decrypted_data[:id]
-      return false
-    elsif (Time.now - decrypted_data[:timestamp]) / 1.hour > 24.0
-      return false
+  def self.validate_token(token)
+    begin
+      decrypted_data = User.decrypt(token)      
+    rescue StandardError => e
```
Comment (Radhikadua123) [inline app/models/user.rb]: Instead of StandardError, can you please check specific type of exception it throws and catch only that.

**Comment by Radhikadua123** [comment]
Thanks for fixing it!

@jywarren Please review the changes.

**Comment by shubhscoder** [comment]
@publiclab/reviewers Please review! Thank you!

**Comment by jywarren** [comment]
This is fantastic. Great work, both of you!!!

**Pull request merged**